File: editorial/chap5-2/prob5-2-2.py
# 入力
N = int(input())
 
# N = 2^k-1 の形で表せるかどうかを調べる
flag = False
for i in range(1, 60):
	if N == (2 ** i) - 1:
		flag = True
 
if flag == True:
	print("Second")
else:
	print("First")

File: editorial/chap5-2/prob5-2-2.c
#include <stdio.h>
#include <stdbool.h>

int main() {
	// 入力
	long long N;
	scanf("%lld", &N);
	
	// N = 2^k-1 の形で表されるかどうかを調べる
	bool flag = false;
	for (int k = 1; k <= 60; k++) {
		if (N == (1LL << k) - 1LL) flag = true;
	}
	
	// 出力
	if (flag == true) printf("Second\n");
	else printf("First\n");
	return 0;
}